build/dart/rules.gni

```diff
@@ -10,21 +10,112 @@ import("//flutter/common/config.gni")
 import("//third_party/dart/build/dart/dart_action.gni")
 import("//third_party/dart/sdk_args.gni")
 
-frontend_server_files =
-    exec_script("//third_party/dart/tools/list_dart_files.py",
-                [
-                  "absolute",
-                  rebase_path("//flutter/flutter_frontend_server"),
-                ],
-                "list lines")
-
-frontend_server_files +=
-    exec_script("//third_party/dart/tools/list_dart_files.py",
-                [
-                  "absolute",
-                  rebase_path("//third_party/dart/pkg"),
-                ],
-                "list lines")
+# Generates a Dart kernel snapshot using flutter_frontend_server.
+#
+# Arguments
+#     main_dart (required):
+#       The Dart entrypoint file.
+#
+#     kernel_output (required):
+#       The path to the output kernel snapshot in the out directory.
+#
+#     package_config (optional):
+#       The path to the package_config.json file.
+#
+#     deps (optional):
+#       Additional dependencies. Dependencies on the frontend server and
+#       Flutter's platform.dill are included by default. This rule creates and
+#       uses a depfile, so listing all Dart sources is not necessary.
+#
+#     extra_args (optional):
+#       Additional frontend server command line arguments.
+template("flutter_frontend_server") {
+  assert(defined(invoker.main_dart), "The Dart test file must be specified.")
+  assert(defined(invoker.kernel_output),
+         "The Dart Kernel file location must be specified.")
+
+  kernel_output = invoker.kernel_output
+
+  common_deps = [ "//flutter/lib/snapshot:strong_platform" ]
+  if (defined(invoker.deps)) {
+    common_deps += invoker.deps
+  }
+
+  extra_args = []
+  if (defined(invoker.extra_args)) {
+    extra_args += invoker.extra_args
+  }
+
+  packages_args = []
+  if (defined(invoker.package_config)) {
+    packages_args += [
+      "--packages",
+      rebase_path(invoker.package_config),
+    ]
+  }
+
+  snapshot_depfile = "$kernel_output.d"
+
+  common_vm_args = [ "--disable-dart-dev" ]
+
+  flutter_patched_sdk = rebase_path("$root_out_dir/flutter_patched_sdk")
+
+  common_args = extra_args + packages_args + [
+                  "--sdk-root",
+                  flutter_patched_sdk,
+                  "--target=flutter",
+                  "--depfile",
+                  rebase_path(snapshot_depfile),
+                  "--output-dill",
+                  rebase_path(invoker.kernel_output),
```

Comment:
Change this to `rebase_path(invoker.kernel_output, root_build_dir)`

This will make Dart generate a depfile containing a relative path to the target.  Without that, GN will not recognize the target in the depfile and will think that this target needs to be rebuilt on every build.

Reply:
Done.